mvn archetype: change to be OSGi java entity

aledsage opened the pull request:
Previously, the generated project included example Java apps and
example main class, with an assembly to produce a stand-alone distort.

This deletes all of those, replacing it with a simple example Java
blueprint. It also includes java-based tests (including for testing
the YAML, and using the YAML test framework).

neykov commented on archetypes/quickstart/src/brooklyn-sample/pom.xml:
```diff
@@ -58,19 +53,30 @@
     <dependency>
-      <!-- this pulls in all brooklyn entities and clouds; 
-           you can cherry pick selected ones instead (for a smaller build) -->
+      <!-- minimal dependency, for writing entities; consider also brooklyn-software-base -->
```
May be list available software modules and what they provide (ex. web, nosql, database, messaging, etc...). How does one figure out which one to add, given a class name (say Tomcat8Server)?

ahgittin replied:
good questions @neykov which should be addressed on the web site, maybe the "code structure" section.  here i think we should link to web site, at most maybe include some commented out sample deps to include.

neykov commented on archetypes/quickstart/src/brooklyn-sample/pom.xml:
```diff
@@ -58,19 +53,30 @@
+      <!-- minimal dependency, for writing entities; consider also brooklyn-software-base -->
       <groupId>org.apache.brooklyn</groupId>
-      <artifactId>brooklyn-all</artifactId>
+      <artifactId>brooklyn-core</artifactId>
```
Or instead leave the -all dependency. Now that this doesn't generate a dist it won't make a difference. The `Import-Package` will be generated on the actually used packages.

neykov commented:
LGTM

ahgittin commented:
minor comments @aledsage then good to merge

aledsage commented:
Thanks @neykov @ahgittin - I've changed it to depend on `brooklyn-all` and changed the comment to the text below. When we have a proper section in the docs to refer to for what each dependency gives, then we can add a specific link.

      <!--
          All Brooklyn dependencies (note that the OSGi imports in the built artifacts will
          only list the packages actually used). It is possible to be far more selective
          about the required dependencies.
      -->

Merging now.
